### PR TITLE
Rubicon Project adapter: Adding support for multiple sizes per slot req

### PR DESCRIPTION
**Pull request by pdezwart**
Rubicon Project cannot currently ingest wxh dims directly. Update maps inbound dims to internal size identifiers and appends to request.

I am horrible at Go, so I will take zero offense to snide remarks about my use of the language. :P

### CODE REVIEW

**Comment by bokelley** [comment]
Pieter - thanks for submitting this! The reason the Travis CI build is failing is because you need to run `go fmt ./...` before you submit. If you don't mind doing that and resubmitting, you should see better results. You can also just run `./validate.sh` to see if it will pass before you commit.

File: adapters/rubicon.go [hunk 1]
```diff
@@ -70,16 +69,63 @@ type rubiconPubExt struct {
 }
 
 type rubiconBannerExtRP struct {
-	SizeID int    `json:"size_id,omitempty"`
-	MIME   string `json:"mime"`
+	SizeID 		int    `json:"size_id,omitempty"`
+	AltSizeIDs 	[]int `json:"alt_size_ids,omitempty"`
```
Comment (bokelley) [inline adapters/rubicon.go]: Does this imply that the Rubicon OpenRTB endpoint is expecting an `alt_size_ids` array?

**Comment by pdezwart** [comment]
Fixed. Don't why I closed and deleted the brain, Ill mark that up as temporary insanity.

File: adapters/rubicon.go (hunk 1 above)
Comment (pdezwart) [inline adapters/rubicon.go]: Yes.

File: adapters/rubicon.go [hunk 2]
```diff
@@ -149,15 +149,15 @@ func (a *RubiconAdapter) Call(ctx context.Context, req *pbs.PBSRequest, bidder *
...
-				primarySizeID = rubiconSizeMap[fmt.Sprintf("%dx%d", unit.Sizes[0].W, unit.Sizes[0].H)]
-				fallthrough
-			case 1:
-				var extraSizes = unit.Sizes[1:]
-				for _, size := range extraSizes {
-					altSizeIDs = append(altSizeIDs, rubiconSizeMap[fmt.Sprintf("%dx%d", size.W, size.H)])
-				}
+		case 0:
+		default:
+			primarySizeID = rubiconSizeMap[fmt.Sprintf("%dx%d", unit.Sizes[0].W, unit.Sizes[0].H)]
```
Comment (asweeney86) [inline adapters/rubicon.go]: We should handle the case where the map lookup fails

File: adapters/rubicon.go [hunk 3]
```diff
@@ -97,13 +143,24 @@ func (a *RubiconAdapter) Call(ctx context.Context, req *pbs.PBSRequest, bidder *
...
 		rpReq.Imp[i].Ext, err = json.Marshal(&impExt)
-		bannerExt := rubiconBannerExt{RP: rubiconBannerExtRP{MIME: "text/html"}}
-		if len(params.Sizes) > 0 {
-			bannerExt.RP.SizeID = params.Sizes[0]
-			rpReq.Imp[i].Banner.Format = nil
-			rpReq.Imp[i].Banner.W = 0
-			rpReq.Imp[i].Banner.H = 0
+
+		primarySizeID := int(0)
+		altSizeIDs := []int{}
```
Comment (asweeney86) [inline adapters/rubicon.go]: We can initialize the map size to the correct size since it is known.  This way we can avoid unnecessary allocations.

Comment (bokelley) [inline adapters/rubicon.go]: that's `make([]int, len(unit.Sizes))` to save you looking up go syntax

**Comment by bokelley** [comment]
I think we should add a test case to rubicon_test.go to make sure these changes work as expected and that future changes don't accidentally break them.

**Comment by bokelley** [comment]
Rewrote in PR #50

**Pull request closed without merging**